### PR TITLE
Replace all BASS fail assertions with describing-exceptions

### PR DESCRIPTION
**Pull request by frenzibyte**
Helps investigating ppy/osu#9045.

Not quite sure about the method naming though.

### CODE REVIEW

File: osu.Framework/Audio/BassUtils.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+using System;
+using ManagedBass;
+using osu.Framework.Logging;
+
+namespace osu.Framework.Audio
+{
+    internal static class BassUtils
+    {
+        private static string getFailString() => $@"BASS failed to initialize with error code {Bass.LastError:D}: {Bass.LastError}.";
```
Comment (peppy) [inline osu.Framework/Audio/BassUtils.cs:12]: any reason this is split out? seems like it would fit in fine inside `CheckFaulted`.

Comment (frenzibyte) [reply osu.Framework/Audio/BassUtils.cs:12]: it is used two times in there, probably should store in a local instead.

**Comment by peppy** [comment]
@frenzibyte made a minor change to better handle initialisation errors. Please check it looks okay.

**Comment by frenzibyte** [comment]
Seems good 👍 , just corrected the fail message I wrote as it was misleading.

**Pull request merged**